docs/src/pages/reference/QueryClient.md

```diff
@@ -203,7 +203,7 @@ This distinction is more a "convenience" for ts devs that know which structure w
 
 ## `queryClient.setQueryData`
 
-`setQueryData` is a synchronous function that can be used to immediately update a query's cached data. If the query does not exist, it will be created. **If the query is not utilized by a query hook in the default `cacheTime` of 5 minutes, the query will be garbage collected**.
+`setQueryData` is a synchronous function that can be used to immediately update a query's cached data. If the query does not exist, it will be created. **If the query is not utilized by a query hook in the default `cacheTime` of 5 minutes, the query will be garbage collected**. To do partial mathing of query keys, you need to use [`queryClient.setQueriesData`](#queryclientsetqueriesdata) instead. 
```

Comment:
I find this more confusing than helpful. Most of the time, you will only update _one_ query. The reference to `partial matching` is also weird. What we could do is:

```
To update multiple queries at once, you can use setQueriesData
```

Reply:
@TkDodo I would not have understood the usage of `setQueriesData` from that sentence. I spent a lot of time figure out how to partially match keys and was giving up on evaluating react-query since such feature did not seem possible.

Reply:
the reason for using "partial" is for searchability. I could not find a way to do what I needed from the documentation since the word was not mentioned anywhere. I only found outdated usages from previous versions which made me think that the feature is completely deprecated.

Reply:
What about: `To update multiple queries at once and partially matching of query keys, you need to use [`queryClient.setQueriesData`](#queryclientsetqueriesdata) instead.` ?

```suggestion
`setQueryData` is a synchronous function that can be used to immediately update a query's cached data. If the query does not exist, it will be created. **If the query is not utilized by a query hook in the default `cacheTime` of 5 minutes, the query will be garbage collected**. To update multiple queries at once and partially matching of query keys, you need to use [`queryClient.setQueriesData`](#queryclientsetqueriesdata) instead. 
```